### PR TITLE
support pip 7, where 'use_wheel' doesn't exist

### PR DESCRIPTION
**Pull request by kouk**

### CODE REVIEW

File: peep.py [hunk 1]
```diff
@@ -104,6 +104,11 @@ def iter(self, ret, *args, **kwargs):
 
     DownloadProgressBar = DownloadProgressSpinner = NullProgressBar
 
+try:
+    from pip.index import FormatControl  # 7.0
```
Comment (edmorley) [inline peep.py]: Change the comment here to ``  # noqa`` to silence the flake8 unused import warning. (And perhaps move the 7.0 comment to the next line).

**Comment by edmorley** [comment]
Ah raced the PR update :-)

**Comment by kouk** [comment]
;-)

**Comment by erikrose** [comment]
Doesn't look like the tests pass under pip 7.0.1. I added 7.0 and 7.0.1 to the tox setup, and this happens:

```
tox -e py27-pip701

======================================================================
ERROR: If a hash matches, peep should do its work and exit happily.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 291, in test_success
    useless==1.0""")
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 267, in install_from_string
    return cls.install_from_path(reqs_path)
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 243, in install_from_path
    local=cls.index_url())
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 106, in run
    shell=True)
  File "/usr/local/Cellar/python/2.7.3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 511, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command 'python /Users/erose/Checkouts/peep/peep.py install -r /var/folders/39/cttgwlvj0rs_73t1z11pltrm0000gn/T/peep-WrGLGa/reqs.txt --index-url http://localhost:8001/' returned non-zero exit status 1

======================================================================
ERROR: Like test_success but with index argument in requirements.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 301, in test_success_req_index
    useless==1.0""")
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 279, in install_from_string_with_index
    return cls.install_from_path_no_index(reqs_path)
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 256, in install_from_path_no_index
    reqs=reqs_path)
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 106, in run
    shell=True)
  File "/usr/local/Cellar/python/2.7.3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 511, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command 'python /Users/erose/Checkouts/peep/peep.py install -r /var/folders/39/cttgwlvj0rs_73t1z11pltrm0000gn/T/peep-fv7p22/reqs.txt --no-index' returned non-zero exit status 1

======================================================================
ERROR: Make sure peep installing a GitHub-sourced tarball installs it,
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 341, in test_upgrade
    index_url=self.index_url()))
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 267, in install_from_string
    return cls.install_from_path(reqs_path)
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 243, in install_from_path
    local=cls.index_url())
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 106, in run
    shell=True)
  File "/usr/local/Cellar/python/2.7.3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 511, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command 'python /Users/erose/Checkouts/peep/peep.py install -r /var/folders/39/cttgwlvj0rs_73t1z11pltrm0000gn/T/peep-199BxH/reqs.txt --index-url http://localhost:8001/' returned non-zero exit status 1

======================================================================
FAIL: Make sure various permutations of inline comments are parsed
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 385, in test_inline_comments
    ['t9XWiL3TRb-ol3d9KXdWaIzwLhs3QsVoheLlwrmW_4I'])
AssertionError: [] != ['t9XWiL3TRb-ol3d9KXdWaIzwLhs3QsVoheLlwrmW_4I']

======================================================================
FAIL: Non-hash or malformed hash comments should be ignored.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 407, in test_non_hash_comments
    eq_(reqs[1]._expected_hashes(), ['abc'])
AssertionError: [] != ['abc']

======================================================================
FAIL: Make sure trailing whitespace is stripped from hashes.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/erose/Checkouts/peep/tests/__init__.py", line 419, in test_whitespace_stripping
    eq_(reqs[0]._expected_hashes(), ['trailing_space_should_be_stripped'])
AssertionError: [] != ['trailing_space_should_be_stripped']

----------------------------------------------------------------------
Ran 9 tests in 3.369s

FAILED (errors=3, failures=3)
```

**Comment by kouk** [comment]
Yes, as @glogiotatidis explained in https://github.com/erikrose/peep/pull/97 there were further incompatibilities. He proposed a fix in that PR, however I went with a different implementation in my last commit here.

**Comment by edmorley** [comment]
@erikrose do you have an opinion re this vs #97?
Would be good to support pip 7.x soon :-)

**Comment by indygreg** [comment]
Some of my servers are starting to pick up pip 7 via system packaging and this bug is causing more and more headaches to work around. Is there any way I can help move this forward?

File: peep.py [hunk 2]
```diff
@@ -104,6 +104,13 @@ def iter(self, ret, *args, **kwargs):
 
     DownloadProgressBar = DownloadProgressSpinner = NullProgressBar
 
+try:
+    from pip.index import FormatControl  # noqa
+    format_control_arg = 'format_control'
```
Comment (erikrose) [inline peep.py]: I suppose these should both be CONSTANTS, or neither should.

File: peep.py [hunk 3]
```diff
@@ -447,23 +456,31 @@ def _path_and_line(self):
     def _expected_hashes(self):
         """Return a list of known-good hashes for this package."""
 
-        def hashes_above(path, line_number):
-            """Yield hashes from contiguous comment lines before line
-            ``line_number``.
+        def get_hash_lists(path):
+            """Yield lists of hashes appearing between non-comment lines.
 
+            The lists will be in order of appearance and, for each non-empty
```
Comment (erikrose) [inline peep.py]: This is really clever!

**Comment by erikrose** [comment]
Belated thanks for this nifty patch, kouk! Peep 2.5 is out, with your fix.

**Pull request merged**